src/__test__/live-preview.test.ts

```diff
@@ -222,6 +229,47 @@ describe("cslp tooltip", () => {
         descPara?.dispatchEvent(hoverEvent);
     });
 
+    test("should redirect to page when edit tag button is clicked for variant field", () => {
+        new LivePreview({
+            enable: true,
+            stackDetails: {
+                apiKey: "sample-api-key",
+                environment: "sample-environment",
+                branch: "dev",
+            },
+        });
+
+        const singularEditButton = document.querySelector(
+            "[data-test-id='cslp-singular-edit-button']"
+        ) as HTMLDivElement;
+
+        const variantTitlePara = document.querySelector(
+            "[data-test-id='title-variant-para']"
+        );
+        const descPara = document.querySelector("[data-test-id='desc-para']");
+
+        jest.spyOn(window, "open").mockImplementation(
+            (url?: string | URL, target?: string, features?: string) => {
+                return {} as Window;
+            }
+        );
+
+        const hoverEvent = new CustomEvent("mouseover", {
+            bubbles: true,
+        });
+
+        variantTitlePara?.dispatchEvent(hoverEvent);
+
+        singularEditButton?.click();
+
+        const expectedRedirectUrl =
+            "https://app.contentstack.com/#!/stack/sample-api-key/content-type/content-type-1/en-us/entry/entry-uid-1/variant/variant-uid-1/edit?branch=dev&preview-field=field-title&preview-locale=en-us&preview-environment=sample-environment";
```

Comment:
yeah ek config variable say hi derive karta tho better hota

Reply:
I'll add it later